backend/infrahub/core/diff/enricher/hierarchy.py

```diff
@@ -1,8 +1,223 @@
+
+        # Check if the parent are already present
+        # If parent is already in the list of node we need to add a relationship
+        # If parent is not in the list of node, we need to add it
+        for node_id, peer_parent in parent_peers.items():
+            # TODO check if we can optimize this part to avoid querying this multiple times
+            node = enriched_diff_root.get_node(node_uuid=node_id)
+            schema_node = self.db.schema.get(
+                name=node.kind, branch=enriched_diff_root.diff_branch_name, duplicate=False
+            )
+            parent_rel = [rel for rel in schema_node.relationships if rel.kind == RelationshipKind.PARENT][0]
+
+            schema_parent = self.db.schema.get(
+                name=peer_parent.peer_kind, branch=enriched_diff_root.diff_branch_name, duplicate=False
+            )
+            parent_side_rel = schema_parent.get_relationship_by_identifier(
+                id=parent_rel.get_identifier(), raise_on_error=False
+            )
+
+            await self._add_parent(
+                enriched_diff_root=enriched_diff_root,
+                node=node,
+                parent_id=str(peer_parent.peer_id),
+                parent_kind=peer_parent.peer_kind,
+                node_side_rel=parent_rel,
+                parent_side_rel=parent_side_rel,
+            )
+
+        if node_parent_with_parent_map:
+            await self._enrich_nodes_with_parent(
+                enriched_diff_root=enriched_diff_root, node_map=node_parent_with_parent_map
+            )
+
+    async def _add_parent(
+        self,
+        enriched_diff_root: EnrichedDiffRoot,
+        node: EnrichedDiffNode,
+        parent_id: str,
+        parent_kind: str,
+        node_side_rel: RelationshipSchema,
+        parent_side_rel: RelationshipSchema | None = None,
+    ) -> EnrichedDiffNode:
+        if not enriched_diff_root.has_node(node_uuid=parent_id):
+            parent = EnrichedDiffNode(
+                uuid=parent_id,
+                kind=parent_kind,
+                label="",
+                action=DiffAction.UNCHANGED,
+                changed_at=Timestamp(),
+            )
+
+            if parent_side_rel:
+                parent.relationships.add(
+                    EnrichedDiffRelationship(
+                        name=parent_side_rel.name,
+                        label=parent_side_rel.label or "",
+                        changed_at=Timestamp(),
+                        action=DiffAction.UNCHANGED,
+                        nodes={node},
+                    )
+                )
+
+            enriched_diff_root.nodes.add(parent)
+
+        else:
+            parent = enriched_diff_root.get_node(node_uuid=parent_id)
+
+        # Add the relationship to the parent
+        if node.has_relationship(name=node_side_rel.name):
+            rel = node.get_relationship(name=node_side_rel.name)
+            if not rel.nodes:
+                rel.nodes.add(parent)
```

Comment:
I might be turned around, but this looks reversed to me
I would expect as to add `node` to the `nodes` attribute of the `rel` on the `parent`, but this seems to be the other way around

Reply:
Done
I also had to modify an issue with the deserializer that wasn't properly creating the relationships for the children nodes ... took me a while to put all the pieces together but I think it should be good now